File: typescript/base-hmi-types/src/screens/widgets/HmiSlider.ts
import { HmiBar } from "./HmiBar.js";
import { HmiObjectType } from "../base/HmiObjectType.js";
import { HmiColor, HmiProperty } from "../base.js";

export class HmiSlider extends HmiBar {
  constructor() {
    super();
    this.hmiObjectType = HmiObjectType.HmiSlider;
  }

  thumbBackgroundColor?: HmiProperty<HmiColor>;
  thumbForegroundColor?: HmiProperty<HmiColor>;
}

File: src/BaseHmiTypes/Screens/Widgets/HmiSlider.cs
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Widgets;

public class HmiSlider : HmiBar
{
    public HmiSlider()
    {
        HmiObjectType = BaseHmiTypes.Screens.Base.HmiObjectType.HmiSlider;
    }

    public HmiProperty<HmiColor>? ThumbBackgroundColor { get; set; }

    public HmiProperty<HmiColor>? ThumbForegroundColor { get; set; }
}
